webservice/pom.xml

```diff
@@ -74,7 +74,29 @@
                 <groupId>org.springframework.boot</groupId>
                 <artifactId>spring-boot-maven-plugin</artifactId>
             </plugin>
+            <plugin>
+                <groupId>com.spotify</groupId>
+                <artifactId>dockerfile-maven-plugin</artifactId>
+                <version>1.4.13</version>
+                <executions>
+                    <execution>
+                        <id>default</id>
+                        <goals>
+                            <goal>build</goal>
+                            <goal>push</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <repository>${project.artifactId}</repository>
+                    <tag>${project.version}</tag>
+                    <buildArgs>
+                        <JAR_FILE>target/${project.build.finalName}.jar</JAR_FILE>
+                    </buildArgs>
+                </configuration>
+            </plugin>
```

Comment:
Just a remark, if you have your Dockerfile in the root of you module, that means that your whole module is considered as the context for the docker build. Basically when you start the docker build all the context is packaged and sent to the docker daemon, which means that all of your file, like source code, target folder, etc.. . This could take a will, that could explain the slowness of your build. There are 2 options to consider as overcome on this issue, 1: use .dockerignore file that tell docker which files should be ignored from the docker context, 2. copy the dockerfile and the jar file from target folder to a common folder in somewhere in the target dir and configure their location in the plugin. This is not a must be done thing, I just wanted to let you know how this works.